src/main/java/spireMapOverhaul/BetterMapGenerator.java

```diff
@@ -36,12 +39,22 @@ private BetterMapGenerator() {
     }
 
     public ArrayList<ArrayList<MapRoomNode>> generate(Random rng, int width, int height, int pathDensity) {
+        if (Settings.isEndless && ActUtil.getRealActNum() == 1) {
```

Comment:
Properly implementing this check -- making endless mode able to see repeat zones -- involved a more robust version of `getRealActNum` that you can see in `ActUtil`. This seemed like a sensible way to do it, since I think endless mode lets you see repeat events too.

There will probably be some janky aspects to doing certain zones multiple times, but endless is always a bit weird, so I figure we fix those as they come up. At least, that seems like it will result in fewer complaints than having endless get no zones at all after seeing each zone once.